specs/themes/ThemeSettings.md

```diff
@@ -0,0 +1,157 @@
+no longer fire.
+
+### Methods
+|Name|Description|
+|-|-|
+|static CreateForWindowId(WindowId)|Creates a ThemeSettings object that will provide theme information for the given WindowId.|
+
+### Events
+|Name|Description|
+|-|-|
+|Changed|Fired when one of the object's properties has changed.|
+
+### Properties
+| Name | Description |
+|-|-|
+| Boolean HighContrast | True if High Contrast mode is enabled on the system. |
+| String HighContrastScheme | If HighContrast is true, a string from the system representing the High Contrast scheme, or style. |
+
+## ThemeSettings.CreateForWindowId static method
+
+``` c#
+static CreateForWindowId(WindowId windowId)
+```
+
+Creates a **ThemeSettings** object that will provide theme information for the given **WindowId**.
+
+The given **WindowId** must represent a window that:
+* is a top-level window,
+* AND is on the same process and thread as the caller.
+
+For example, you can get a **WindowId** from the **ContentEnvironment.AppWindowId** property.  You can get a **ContentEnvironment**
+object from a **XamlRoot** object.
+
+## ThemeSettings.Changed event
+
+```c#
+event TypedEventHandler<ThemeSettings, Object> Changed
+```
+
+Fired when one of the object's properties has changed.
+
+The **Changed** event will only fire while the **ThemeSettings** object is alive, so apps must ensure
+they hold a reference to the object as long as they want to receive the event.
+
+The **Changed** event will only fire as long as the window represented by **WindowId** is alive (has not yet received `WM_NCDESTROY`).
+
+# API Details
+
+_Spec note: this is a new namespace, **Microsoft.UI.System**.
+The predecessor for this type is `AccessibilitySettings`,
+which is in the `Windows.UI.ViewManagement` namespace,
+which is all UWP types we don't plan to carry into Desktop.
+This is a UI type that doesn't fit into any of the children of the `UI` namespace.
+We anticipate more types here, but don't have any current plans._
+
+
+``` c# (really midl3)
+namespace Microsoft.UI.System
+{
+  runtimeclass ThemeSettings 
+  {
+    static ThemeSettings CreateForWindowId(WindowId windowId);
+
+    event TypedEventHandler<ThemeSettings, Object> Changed;
+
+    Boolean HighContrast { get; }
+    String HighContrastScheme { get; } 
+  }
+}
+```
+
+# Appendix
+
+## Open Issues and Implementation Issues
+In the experimental API today, HighContrastScheme returns a stale value if HighContrast is off.  We should change this to return
+the empty string.
+
+It seems easy to misuse this API by accidentally allowing the ThemeSettings object to go out of scope, which will mean the event
+never fires.
+
+In the original plan we weren't going to add HighContrast or HighContrastScheme.  Do we need them?  Should we remove?
+
+Note that today in the code the **Changed** object is called **HighContrastChanged** because it currently only applies to HighContrast.
```

Comment:
Does this mean that changing HighContrastScheme from Aquatic to Dusk wouldn't proc a ThemeSettings::Changed event?  That sounds like a bug whether we have a peer Boolean or not.  UX trying to respect the current scheme will be left out of date.

Reply:
I'm hoping this just means "only applies to HighContrast *(state or theme)*." I.e., "Since all of the properties on this object are HC-related, the event name is currently HCChanged. But this new API is going to use the Changed event to notify for any changes to the object, even future ones that may not be HC-related."

Reply:
Sorry for the confusion.  Raymond is correct about what I meant.
(Changing HighContrastScheme from Aquatic to Dusk will raise the ThemeSettings.Changed event).

Reply:
Discussion: The `Changed` event is following the aggregate change event pattern here, meaning that when any of it changes the one event raises. This wraps WM_THEMECHANGED, so being more granular such as HighContrastChanged would actually be difficult to do, and the aggregate pattern is a not-uncommon one today.

Reply:
I've removed this from the appendix.